### DISCUSSION TITLE
How to store a buffer with .moveToDisk

### DISCUSSION THREAD

**Discussion by Tob0t** [adonisjs/core#4273] [Help] [answered]
Hi,
I'm currently storing my images with the `.moveToDisk()` method which is quite cosy to do automatic the filename-generation for me.

```
 public async storeImage(image: MultipartFileContract, folder: string) {
    await image.moveToDisk(folder)
    return image.fileName
  }
  ```
  
  Now I also want to resize the image where I decided to use the sharp lib. However, now I am confused how to use the `moveToDisk` method from a given buffer. Is there any possibility or do I need to fall back to `Drive.put()`?
  
  ```
 public async storeImage(image: MultipartFileContract, folder: string) {
    // resize image
    const resizedImageData = await sharp(image.tmpPath).resize(500).toBuffer()
    await resizedImageData.moveToDisk(folder)
    return resizedImageData.fileName
  }
  ```

**Comment by thetutlage** [comment, marked as answer]
The `image.moveToDisk` is a method that exists on the `MultipartFile` instance. Now, once you resize this file and get a buffer, you are not longer using the multipart file, therefore you will have to use `Drive` APIs directly.

You can use the [`cuid` helper](https://docs.adonisjs.com/guides/helpers#cuid) to generate a random name.
